lib/cozy-light.js

```diff
@@ -1,1296 +1,93 @@
+  status: 'stopped',
+  init: function( program ){
+    cozyLight.logger.info('initializing..');
+    cozyLight.npmHelpers.initialWd = process.cwd();
+    var defaultHome = pathExtra.join(pathExtra.homedir(), '.cozy-light');
+    cozyLight.configHelpers.init(program.home || defaultHome);
+    cozyLight.configHelpers.setMainAppPort(program.port || 19104);
+    process.chdir( cozyLight.configHelpers.getHomePath() );
+    cozyLight.logger.info('Home is ' + cozyLight.configHelpers.getHomePath());
+    cozyLight.db = new Pouchdb('cozy');
+    cozyLight.pluginHelpers.loadAll(program);
+    // Reload apps when file configuration is modified
+    cozyLight.configHelpers.watcher.on(function configChanged () {
+      cozyLight.logger.info('Config changed...');
+      if ( cozyLight.status === 'started' ) {
+        cozyLight.logger.info('...Restarting...');
+        cozyLight.actions.restart(program);
       }
-
-      LOGGER.info('Stopping plugins...');
-      pluginHelpers.stopAll(function (err) {
-        if (err) {
-          LOGGER.error('An error occured while stopping plugins');
-          LOGGER.raw(err);
-        }
-
-        LOGGER.info('Stopping server...');
-        mainAppHelper.stop(function(){
-          port = defaultAppsPort;
-          LOGGER.info('\t' + symbols.ok + '\tmain server');
-          if (callback) { callback(); }
-        });
-      });
     });
   },
-
-  /**
-   * Stops application server, its apps, and its plugins,
-   * reload everyone, then restart everyone
-   */
-  restart: function (callback) {
-    LOGGER.info('Restarting...');
-    actions.stop(function(){
-      configHelpers.loadConfigFile();
-      loadedPlugins = {};
-      loadedApps = {};
-      routes = {};
-      LOGGER.info('Starting all apps...');
-      actions.start(program, function(){
-        LOGGER.info('...Cozy Light was properly restarted.');
-        if (callback) { callback(); }
-      });
-    });
-  },
-
   /**
    * Manage properly exit of the process when SIGINT signal is triggered.
    * It asks to every plugin to end properly.
    */
-  exit: function (callback) {
-    actions.stop(function endProcess (err) {
+  stop: function( callback ){
+    cozyLight.actions.stop(function endProcess (err) {
       if (err) {
-        LOGGER.error('Cozy Light was not properly terminated.');
+        cozyLight.logger.error('Cozy Light was not properly terminated.');
       } else {
-        LOGGER.info('Cozy Light was properly terminated.');
+        cozyLight.logger.info('Cozy Light was properly terminated.');
       }
 
-      if (db) {
+      var doneCallback = function(){
+        cozyLight.configHelpers.watcher.release();
+        cozyLight.nodeHelpers.invoke(callback);
+      };
+
+      if (cozyLight.db) {
         try {
-          if(!db._closed) {
-            db.close();
+          /*eslint-disable */
+          if(!cozyLight.db._closed) {
+            /*eslint-enable */
+            return cozyLight.db.close(doneCallback);
           }
```

Comment:
In the case you mention the execution continues unitl L57. It is intended to avoid else and prefer return callback(); syntax. It helps to reduce braces, thus, complexity during the reading phase. But yet, it is less expressive because more natural.